sql/core/src/main/scala/org/apache/spark/sql/jdbc/OracleDialect.scala

```diff
@@ -61,6 +61,34 @@ private case class OracleDialect() extends JdbcDialect with SQLConfHelper with N
       } else {
         super.visitAggregateFunction(funcName, isDistinct, inputs)
       }
+
+    private def compareBlob(lhs: Expression, operator: String, rhs: Expression): String = {
+      val l = inputToSQL(lhs)
+      val r = inputToSQL(rhs)
+      val op = if (operator == "<=>") "=" else operator
+      val compare = s"DBMS_LOB.COMPARE($l, $r) $op 0"
+      if (operator == "<=>") {
+        s"(($l IS NOT NULL AND $r IS NOT NULL AND $compare) OR ($l IS NULL AND $r IS NULL))"
```

Comment:
This is dirty but I don't have better ideas...

Reply:
Yes, it's dirty. Fortunately, we only found Oracle requires a special function to compare blob values.

Reply:
Even if we change signature of visitBinaryComparison to make the override better, we still cannot get rid of these dirty rewrite of `<=>` and `DBMS_LOB.COMPARE`.